150627370 Help Centre

emugaya opened the pull request:
## What does this PR do?
1. Adds Link to helpcenter on the homepage
2. Adds Frequently Asked Questions to Healthchecks
3. Adds Videos to Healhchecks
4. Adds Livechat to Healthchecks

## Description of Task to be completed?
1. Help center has been added to healthchecks to provide resources for new users to refer to. These are Video and Frequently Asked Questions.
2. Videos can only be added by an admin from the django admin. The admin is able to add a title of the video and the URL to either `Youtube` or `Vimeo`.
3. Frequently asked questions can only be added from the django admin interface by a super user by providing a question and answer.
4. A Clickdesk chat script has been added to the `base.html` file. This enables a visitor to chat with any of the healthchecks agents on Clickdesk

## Completed tests in the hc/front/tests folder
The tests are in a file `test_for_chat_faqs_and_videos.py`

## What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/150627370

emugaya committed:
Feature(150627370)Help Center. Added Faq, and Video models to admin site

emugaya committed:
Feature(150627370)Help Center. Added Faq, and Video models to admin site

emugaya committed:
Feature(150627370)Help Center. Added Faq, and Video models

emugaya committed:
Feature(150627370)Help Center. Added Faq, and Video models

emugaya committed:
Feature(150627370)Help Center. Adding urls to Helpcenter Videos, Chat and FAQs

emugaya committed:
Feature(150627370)Help Center. Adding urls to Helpcenter Videos, Chat and FAQs

emugaya committed:
Feature(150627370)Help Center. Created views for displaying, videos, faqs, and helpcenter pages

emugaya committed:
Feature(150627370)Help Center. Created views for displaying, videos, faqs, and helpcenter pages

emugaya committed:
Feature(150627370)Help Center. Added django-video-embed package

emugaya committed:
Feature(150627370)Help Center. Added django-video-embed package

emugaya committed:
Feature(150627370)Help Center. Adding Helpcenter and it's link

emugaya committed:
Feature(150627370)Help Center. Adding Helpcenter and it's link

emugaya committed:
Feature(150627370)Help Center Templates for Video and FAQ

emugaya committed:
Feature(150627370)Help Center Templates for Video and FAQ

emugaya committed:
Feature(150627370)Help Center. Adjusted location Back to Helpcenter Button

emugaya committed:
Feature(150627370)Help Center. Adjusted location Back to Helpcenter Button

emugaya committed:
Feature(150627370)Help Center. Add django-video-embed to requirements.txt

emugaya committed:
Feature(150627370)Help Center. Add django-video-embed to requirements.txt

emugaya committed:
Fix Merge Conflicts in hc/settings.py and requirements.txt

emugaya committed:
Fix Merge Conflicts in hc/settings.py and requirements.txt

emugaya committed:
Fix merge conflicts

emugaya committed:
Testing page has content that was posted

emugaya committed:
Feature(150627370)Added Chat Functionality
- A script from helpdesk has been copied to base.html

emugaya committed:
Feature(150627370) Help Center
- Added Tests for chat, creating and displaying FAQs and Videos

emugaya committed:
Feature(150627370)Help Center Added link to blog posts

emugaya committed:
Fix merge conflicts between development and 150627370-help-center

emugaya committed:
Fix merge conflicts in requirements.txt

emugaya committed:
Fix Merge Conflicts in hc/front/views.py

emugaya committed:
Fix merge conflicts

emugaya committed:
Fixes Indentation in hc/front/views.py

emugaya committed:
Fixes syntax error in hc/front/views.py

margierain commented on hc/blogs/tests/test_blog_post.py:
```diff
@@ -29,6 +29,8 @@ def test_create_blog_post(self):
 		self.assertEquals(len(post), 3)
 		response = self.client.get(self.URL+'post/' + post_id +'/')
 		self. assertEquals(response.status_code, 200)
+		print(type(response.content))
```
Remove print function

emugaya committed:
Feature(150627370)Help Center
- Added helpful messages to be displayed to user when there are no blogs, faqs and Videos to display
- Added Filtering blog posts by category/ tag
- Fixed pagination for. We are now to view number of pages and naigate to multiple pages
- Removed print statement from blog test file.
- Added docstrings to Blog, and Help Center models and views

emugaya committed:
Merge branch 'development' into 150627370-help-center
